presto-release-tools/src/test/java/com/facebook/presto/release/tasks/TestFinalizeReleaseTask.java

```diff
@@ -134,4 +192,22 @@ private static void assertCommands(CommandLogger commandLogger)
                         "mvn release:clean",
                         "git push upstream -u release-0.231:release-0.231 --tags"));
     }
+
+    private static void assertCommandsHotFix(CommandLogger commandLogger)
+    {
+        assertEquals(
+                commandLogger.getCommands(),
+                ImmutableList.of(
+                        "git status -s",
+                        "git checkout master",
+                        "git pull --ff-only upstream master",
+                        "git fetch upstream",
+                        "git tag",
+                        "git ls-remote --heads upstream release-0.231",
+                        "git branch -D release-0.231",
```

Comment:
I guess I am not familiar with how hotfixes should work, but we delete the branch instead of replacing it or adding commits to it? What is the point of having a release branch for a version if it only points to the last hotfix?

What I expect is a branch to be for a something that is going to receive several patch releases and the releases are tagged on that branch.

The branch is there to facilitate the backporting of changes to the release without the other changes that may have already occurred on master.

Reply:
The "delete branch" command is not specific to the hotfix process. When finalizing the release, we always attempt to delete the local release branch, if there is one, and capture/ignore the failure if there is none.

That way, local commits are discarded and the next commands always creates a new local release branch that are synced with `upstream`:
```
"git checkout -b release-0.231 upstream/release-0.231"
```

After getting the latest version of the release branch, the pom version on the release branch will be:
- `0.231-SNAPSHOT` if we haven't perform the release yet.
  - We push a commit to update the version to `0.231`, tag it with `0.231`.
  - We push another commit to update the version to `0.231.1-SNAPSHOT`.
- `0.231.1-SNAPSHOT` if we have already released `0.231`, and is requesting a hotfix.
  - We push a commit to update the version to `0.231.1`, tag it with `0.231.1`.
  - We push another commit to update the version to `0.231.2-SNAPSHOT`.